server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -89,6 +90,21 @@ public abstract class TransportReplicationAction<
     ReplicaRequest extends ReplicationRequest<ReplicaRequest>,
     Response extends ReplicationResponse> extends TransportAction<Request, Response> {
 
+    protected enum ActionFlags {
+        /**
+         * Bypass queue-length checks when executing on the primary
+         */
+        ForceExecutionOnPrimary,
+        /**
+         * Sync the global checkpoint to the replicas after the action completes successfully
+         */
+        SyncGlobalCheckpointAfterOperation,
+        /**
+         * Prevent the replica action being rejected by the circuit breakers if the primary action succeeded
+         */
+        BypassCircuitBreakerOnReplica
```

Comment:
Not entirely sold on the name here. "bypass" could be interpreted as don't add to the circuit breaker at all, when it actually means add, but don't fail here. Happy for any suggestion of a better name.